group/assignment6/parking_class.py

```diff
@@ -0,0 +1,54 @@
+class Parking:
+    """
+    Args:
+        constraints: {parking_lot: (permitted cars)}
+    """
+    def __init__(self, state):
+        self._state = state
+        self._inverted_state = self._compute_inverted_state()
+
+    def __len__(self):
+        return len(self._state)
+
+    def _compute_inverted_state(self):
+        inverted_state = list(range(len(self._state)))
+        for lot, car in enumerate(self._state):
+            inverted_state[car] = lot
+        return inverted_state
+
+    def get_car(self, lot):
```

Comment:
we can rewrite it in this way:
@property
def car(self, lot):
         return self._state[lot]
but I dont tonic that it will be useful, I don't know best practices about properties in python, I think it is useful for set methods but not for get